### PR TITLE
fix inlay hints for template literal type

### LINKED ISSUE #56544: Inlay hints causing a crash on TemplateLiteralType
### 🔎 Search Terms

"Inlay hints", "TemplateLiteralType"

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions master and 5.2.2

### ⏯ Playground Link

_No response_

### 💻 Code

```ts
declare function getTemplateLiteral1(): `${string},${string}`;
const lit1 = getTemplateLiteral1();
```

### 🙁 Actual behavior

it does not display the type of `lit1`
image

### 🙂 Expected behavior

it should display the type of `lit1`
image

### Additional information about the issue

`TemplateLiteralType` is not supported by `visitForDisplayParts` yet.
https://github.com/microsoft/TypeScript/actions/runs/6755514395/job/18364019347?pr=56309#step:5:50

### PR DESCRIPTION
**Pull request by Zzzen**
fix #56544

### CODE REVIEW

**Comment by sandersn** [comment]
Can you explain what the bug was, what changed, and how it fixes the bug? It would be nice to have a bug tracking this as well.

File: tests/baselines/reference/inlayHintsInteractiveVariableTypes3.baseline [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      "text": "`start${"
+    },
+    {
+      "text": "string"
+    },
+    {
+      "text": "}\\${,$${"
+    },
+    {
+      "text": ""
```
Comment (MariaSolOs) [inline tests/baselines/reference/inlayHintsInteractiveVariableTypes3.baseline:84]: This looks wrong. I don't think we should generate empty display parts like these.

Comment (Zzzen) [reply tests/baselines/reference/inlayHintsInteractiveVariableTypes3.baseline]: Rpdated.

**Comment by Zzzen** [comment]
> Can you explain what the bug was, what changed, and how it fixes the bug? It would be nice to have a bug tracking this as well.

I've added an issue to track the bug mentioned. https://github.com/microsoft/TypeScript/issues/56544

File: tests/cases/fourslash/inlayHintsInteractiveVariableTypes3.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/// <reference path="fourslash.ts" />
```
Comment (MariaSolOs) [inline tests/cases/fourslash/inlayHintsInteractiveVariableTypes3.ts:1]: For smoother maintenance, could you please use a more descriptive name for this test file? E.g. `inlayHintsInteractiveTemplateLiteralTypes.ts`.

**Review by MariaSolOs — APPROVED** [review]

**Pull request merged**